mantidproject/mantid
# Fix documentation build for published docs

**Description of work.**

Fixes issue introduced by changes in #29684. Diagrams had been accidentally disabled and the algorithm caching mechanism caused erroneous values in the properties tables. These have been fixed.

**To test:**

* Build the docs locally and check the page for `SANSILLReduction`. The properties table should not include many workspace names etc.
* Code review the buildscript and ensure the option for diagram generation now matches that in `docs/CMakeLists.txt`

*There is no associated issue.*

*This does not require release notes* because **fixes an internal issue.**



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
